src/test/ui/associated-types/cache/chrono-scan.rs

```diff
@@ -21,9 +21,9 @@ pub fn parse<'a, I>(mut s: &str, items: I) -> ParseResult<()>
         ($e:expr) => ({ let (s_, v) = $e?; s = s_; v })
     }
     #[allow(deprecated)]
-    let offset = try_consume!(timezone_offset_zulu(s.trim_left(), colon_or_space));
+    let offset = try_consume!(timezone_offset_zulu(s.trim_start(), colon_or_space));
```

Comment:
What deprecated about this line?

Reply:
```
---- [ui] ui/associated-types/cache/chrono-scan.rs stdout ----
normalized stderr:
warning: use of deprecated item 'try': use the `?` operator instead
  --> $DIR/chrono-scan.rs:21:39
   |
LL |         ($e:expr) => ({ let (s_, v) = try!($e); s = s_; v })
   |                                       ^^^^^^^^
LL |     }
LL |     let offset = try_consume!(timezone_offset_zulu(s.trim_start(), colon_or_space));
   |                  ------------------------------------------------------------------ in this macro invocation
   |
   = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'try': use the `?` operator instead
  --> $DIR/chrono-scan.rs:21:39
   |
LL |         ($e:expr) => ({ let (s_, v) = try!($e); s = s_; v })
   |                                       ^^^^^^^^
...
LL |     let offset = try_consume!(timezone_offset_zulu(s.trim_start(), colon_or_space));
   |                  ------------------------------------------------------------------ in this macro invocation
```
I had this error message. I don't understand it fully, but I assumed it somehow involed the `try!` macro and needs an `#[allow(deprecated)]` as well

Reply:
the `trim_left` to `trim_start` was just an mistake, because my master wasn't in sync with upstream

Reply:
I mean should we keep the `#[allow(deprecated)]` attribute?

Reply:
>@BO41 You can use #[allow(...)] in more places for now. :)

Well @Centril suggested it.
If you have a real solution, feel free to open a follow up PR without `allow` :wink:

Reply:
You can use `#[allow(deprecated)]` inside the macro rather than where it is called.

Reply:
I just noticed that I _already_ replaced the `try!` call in the macro. So the macro call doesn't have anything to do with this.
So I should neither need `try!` nor `#[allow(deprecated)]` :man_facepalming: